api/client/client_test.go

```diff
@@ -778,3 +778,191 @@ func TestAccessRequestDowngrade(t *testing.T) {
+	require.NoError(t, err)
+	require.Equal(t, types.RequireSessionMFA, roleResp.GetOptions().RequireMFAType)
+
+	// GetRoles should set "RequireMFAType" on the received roles if empty
+	role.Spec.Options.RequireMFAType = ""
+	role.Spec.Options.RequireSessionMFA = true
+	rolesResp, err := clt.GetRoles(ctx)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(rolesResp))
+	require.Equal(t, types.RequireSessionMFA, rolesResp[0].GetOptions().RequireMFAType)
+
+	// GetCurrentUserRoles should set "RequireMFAType" on the received roles if empty
+	role.Spec.Options.RequireMFAType = ""
+	role.Spec.Options.RequireSessionMFA = true
+	rolesResp, err = clt.GetCurrentUserRoles(ctx)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(rolesResp))
+	require.Equal(t, types.RequireSessionMFA, rolesResp[0].GetOptions().RequireMFAType)
+}
+
+type mockAuthPreferenceServer struct {
+	*mockServer
+	pref *types.AuthPreferenceV2
+}
+
+func newMockAuthPreferenceServer() *mockAuthPreferenceServer {
+	m := &mockAuthPreferenceServer{
+		mockServer: &mockServer{
+			grpc:                           grpc.NewServer(),
+			UnimplementedAuthServiceServer: &proto.UnimplementedAuthServiceServer{},
+		},
+	}
+	proto.RegisterAuthServiceServer(m.grpc, m)
+	return m
+}
+
+func startMockAuthPreferenceServer(t *testing.T) string {
+	l, err := net.Listen("tcp", "")
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, l.Close()) })
+	go newMockAuthPreferenceServer().grpc.Serve(l)
+	return l.Addr().String()
+}
+
+func (m *mockAuthPreferenceServer) GetAuthPreference(ctx context.Context, _ *empty.Empty) (*types.AuthPreferenceV2, error) {
+	if m.pref == nil {
+		return nil, trace.NotFound("not found")
+	}
+	return m.pref, nil
+}
+
+func (m *mockAuthPreferenceServer) SetAuthPreference(ctx context.Context, pref *types.AuthPreferenceV2) (*empty.Empty, error) {
+	m.pref = pref
+	return &empty.Empty{}, nil
+}
+
+// Test that client will perform properly with an old server
+// DELETE IN 13.0.0
+func TestSetAuthPreferenceRequireSessionMFABackwardsCompatibility(t *testing.T) {
+	ctx := context.Background()
+	addr := startMockAuthPreferenceServer(t)
+
+	// Create client
+	clt, err := New(ctx, Config{
+		Addrs: []string{addr},
+		Credentials: []Credentials{
+			&mockInsecureTLSCredentials{}, // TODO(Joerger) replace insecure credentials
+		},
+		DialOpts: []grpc.DialOption{
+			grpc.WithTransportCredentials(insecure.NewCredentials()), // TODO(Joerger) remove insecure dial option
+		},
+	})
+	require.NoError(t, err)
+
+	pref := &types.AuthPreferenceV2{
+		Metadata: types.Metadata{
+			Name: "one",
+		},
+	}
+
+	// Upsert should set "RequireSessionMFA" on the provided auth pref if "RequireMFAType" is set
+	pref.Spec.RequireMFAType = types.RequireSessionMFA
+	pref.Spec.RequireSessionMFA = false
+	err = clt.SetAuthPreference(ctx, pref)
+	require.NoError(t, err)
+	require.Equal(t, pref.Spec.RequireMFAType.IsSessionMFARequired(), pref.Spec.RequireSessionMFA)
```

Comment:
```suggestion
	// SetAuthPreference should set "RequireSessionMFA" on the provided auth pref if "RequireMFAType" is set
	pref.Spec.RequireMFAType = types.RequireSessionMFA
	pref.Spec.RequireSessionMFA = false
	err = clt.SetAuthPreference(ctx, pref)
	require.NoError(t, err)
	require.True(t, pref.Spec.RequireSessionMFA)
```

Same comment here. Just assert it's equal to `true`.